core/tests/protractor_utils/ProfilePage.js

```diff
@@ -62,14 +65,20 @@ var ProfilePage = function() {
     expect(numInterests).toEqual(expectedInterests.length);
 
     var interestTexts = await interests.map(async function(interestElem) {
-      return await interestElem.getText();
+      await waitFor.visibilityOf(
+        interestElem,
+        'interestElem is taking too Long to appear');
+      expect(await interestElem.getText()).toMatch(expectedInterests);
```

Comment:
The `return await interestElem.getText();` is still correct. `expectedInterests` is a list, so it will never be equal to `interestElem.getText()`. What we're trying to do here is get a list (`interestTexts`) of all the interest element texts. Then we'll compare that list to `expectedInterests` below